test/e2e/customer_admin_reader_test.go

```diff
@@ -0,0 +1,59 @@
+//+build e2e
+
+package e2e
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = Describe("Openshift on Azure customer-cluster-reader e2e tests [CustomerAdmin]", func() {
+	defer GinkgoRecover()
+
+	It("should not read nodes", func() {
+		_, err := creader.kc.CoreV1().Nodes().Get("master-000000", metav1.GetOptions{})
+		Expect(kerrors.IsForbidden(err)).To(Equal(true))
+	})
+
+	It("should have read-only on all non-infrastructure namespaces", func() {
+		// create project as enduser
+		namespace := nameGen.generate("e2e-test-")
+		c.createProject(namespace)
+		defer c.cleanupProject(10 * time.Minute)
+
+		err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			rb, err := creader.kc.CoreV1().Namespaces().Get(c.namespace, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
```

Comment:
Return nil only when the error is not found

Reply:
Will return err.

Reply:
The issue here is a race condition.  We are fetching the project immediately after create was called.  This causes an error.  The error could be a few:
- does not exist
- forbidden (reconciler has not processed this namespace)